src/myrtlespeech/model/rnn.py

```diff
@@ -11,6 +12,37 @@ class RNNType(IntEnum):
     BASIC_RNN = 2
 
 
+#: The type of an :py:class:`RNN` hidden state.
+#:
+#: Depending on the :py:class:`RNN`'s :py:class:`RNNType`, the hidden state
+#: will either be a length 2 Tuple of :py:class:`torch.Tensor`s or a single
+#: :py:class:`torch.Tensor` (see :py:class:`torch.nn` documentation for more
+#: information).
+RNNState = TypeVar("RNNState", torch.Tensor, Tuple[torch.Tensor, torch.Tensor])
+
+
+#: The type of the sequence data input to a :py:class:`RNN`.
+#:
+#: The :py:class:`RNN` input ``x`` type is polymorphic: either it is a
+#: Tuple ``x = (inp, hid)`` or it is of the form: ``x = inp`` where ``inp``
+#: is the network input, a :py:class:`torch.Tensor`) with size
+#: ``[seq_len, batch, in_features]`` or ``[batch, seq_len, in_features]``
+#: depending on whether ``batch_first=True`` and ``hid`` is the
+#: :py:class:`RNN` hidden state of type :py:class:`RNNType`.
```

Comment:
Minor edit suggestions: 

Remove reference to `x` and just discussing the form of the input. It is confusing as to what `x` is without looking at the `RNN` implementation.

There's a closing parenthesis instead of a comma?

`RNNType` should be `RNNState` I think. 

Taking all these into account, perhaps:

```
#: The type of the sequence data input to a :py:class:`RNN`.
#:
#: The :py:class:`RNN` input type is polymorphic: either it is a Tuple
#: ``(inp, hid)`` or it is of the form ``inp``, where ``inp`` is the network
#: input, a :py:class:`torch.Tensor`, with size ``[seq_len, batch,
#: in_features]`` or ``[batch, seq_len, in_features]`` depending on whether
#: ``batch_first=True`` and ``hid`` is the :py:class:`RNN` hidden state of type
#: :py:class:`RNNState`.
```

Reply:
Yep good point. Updated.